tests/cupyx_tests/scipy_tests/stats_tests/test_stats_py.py

```diff
@@ -0,0 +1,178 @@
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_multi_dim(self, xp, scp, dtype):
+        x = testing.shaped_random((3, 4, 5, 7), xp, dtype=dtype)
+        y = testing.shaped_random((3, 4, 1, 1), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=atol, rtol=rtol)
+    def test_zmap_multi_dim_2(self, xp, scp, dtype):
+        x = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=dtype)
+        y = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-3, rtol=1e-3)
+    def test_zmap_multi_dim_2_float16(self, xp, scp):
+        x = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=xp.float16)
+        y = testing.shaped_random((4, 4, 5, 6, 2), xp, dtype=xp.float16)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_with_axis(self, xp, scp, dtype):
+        x = testing.shaped_random((2, 3), xp, dtype=dtype)
+        y = testing.shaped_random((1, 3), xp, dtype=dtype)
+        return scp.stats.zmap(x, y, axis=1)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_with_axis_ddof(self, xp, scp, dtype):
+        x = testing.shaped_random((2, 3), xp, dtype=dtype)
+        y = testing.shaped_random((1, 3), xp, dtype=dtype)
+        return scp.stats.zmap(x, y, axis=1, ddof=2)
+
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_empty(self, xp, scp, dtype):
+        x = xp.array([], dtype=dtype)
+        y = xp.array([1, 3, 5], dtype=dtype)
+        return scp.stats.zmap(x, y)
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_nan_policy_propagate(self, xp, scp, dtype):
+        x = xp.array([4.0, 1.0, 1.0, cupy.nan], dtype=dtype)
+        y = xp.array([cupy.nan, -4.0, -1.0, -5.0], dtype=dtype)
+        return scp.stats.zmap(x, y, nan_policy='propagate')
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_nan_pocily_omit(self, xp, scp, dtype):
+        x = xp.array([4.0, 1.0, 1.0, cupy.nan], dtype=dtype)
+        y = xp.array([cupy.nan, -4.0, -1.0, -5.0], dtype=dtype)
+        return scp.stats.zmap(x, y, nan_policy='omit')
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_zmap_nan_policy_omit_axis_ddof(self, xp, scp, dtype):
+        x = xp.array([4.0, 1.0, 1.0, cupy.nan], dtype=dtype)
+        y = xp.array([cupy.nan, -4.0, -1.0, -5.0], dtype=dtype)
+        return scp.stats.zmap(x, y, axis=0, ddof=1, nan_policy='omit')
+
+    @testing.for_dtypes('fdFD')
+    def test_zmap_nan_policy_raise(self, dtype):
```

Comment:
Let's test that both scipy and cupy raise an error.
```py
for xp, scp in [(numpy, scipy), (cupy, cupyx.scipy)]:
    ....
    with pytest.raises(ValueError):
        ...
```

Reply:
Yes, sure! Update it, thanks!